bftengine/src/bftengine/PersistentStorageImp.hpp

```diff
@@ -20,8 +20,13 @@
 namespace bftEngine {
 namespace impl {
 
-enum ConstMetadataParameterIds {
-  VERSION_PARAMETER = 0,
+// Make a reservation for future params
+const uint16_t reservedParamsNum = 50;
```

Comment:
For simple parameters, a gap between ConstMetadataParameterIds and WinMetadataParameterIds solves the problem. The whole number of parameters - MAX_METADATA_PARAMS_NUM is 10000, so we have a lot of space for new parameters at the end, too. But windows and descriptors structure is too complicated to be resolved in a similar way. In case we need to add one parameter to each window element, 300 new messages should be added and in very specific places. I don't see how this problem could be resolved without a database converter.

Reply:
Ok. We'll have to discuss this more offline.